migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs

```diff
@@ -549,6 +534,47 @@ impl<'schema> SqlSchemaDiffer<'schema> {
     }
 }
 
+fn push_previous_usages_as_defaults_in_altered_enums(differ: &SqlSchemaDiffer<'_>, alter_enums: &mut [AlterEnum]) {
+    for alter_enum in alter_enums {
+        let mut previous_usages_as_default = Vec::new();
+
+        let enum_names = differ.schemas.enums(&alter_enum.index).map(|enm| enm.name());
+
+        for table in differ.dropped_tables() {
+            for column in table
+                .columns()
+                .filter(|col| col.column_type_is_enum(enum_names.previous()) && col.default().is_some())
+            {
+                previous_usages_as_default.push(((column.table().table_index(), column.column_index()), None));
+            }
+        }
+
+        for tables in differ.table_pairs() {
+            for column in tables
+                .dropped_columns()
+                .filter(|col| col.column_type_is_enum(enum_names.previous()) && col.default().is_some())
+            {
+                previous_usages_as_default.push(((column.table().table_index(), column.column_index()), None));
+            }
+
+            for columns in tables.column_pairs().filter(|col| {
+                col.previous.column_type_is_enum(enum_names.previous()) && col.previous.default().is_some()
+            }) {
+                let next_usage_as_default = Some(&columns.next)
+                    .filter(|col| col.column_type_is_enum(enum_names.next()) && col.default().is_some())
+                    .map(|col| (col.table().table_index(), col.column_index()));
+
+                previous_usages_as_default.push((
+                    (columns.previous.table().table_index(), columns.previous.column_index()),
+                    next_usage_as_default,
+                ));
+            }
+        }
+
+        alter_enum.previous_usages_as_default = previous_usages_as_default;
+    }
+}
```

Comment:
do we want this kind of code? I think we could do better, thinking about how.

Reply:
You mean mutating things in a weird order? I also find it a bit stressful to follow...

Reply:
`I think the order is very straightforward (we just accumulate in one vector), it's the repetition that I find distressing.`

Reply:
Yeah, I can't figure out how much repetition there is already, but this looks like an important fix to get to the release. Do you want to hurry it, or do another round of refactoring still?

Reply:
Nah I can't get it better in this PR, it has to be more systematic — I'll explore that in the learning day.